src/stdlib_stats_distribution_normal.fypp

```diff
@@ -0,0 +1,332 @@
+        ${t1}$, intent(in) :: loc, scale
+        integer, intent(in) :: array_size
+        ${t1}$ :: res(array_size)
+        ${t1}$, parameter  ::  r = 3.442619855899_${k1}$, rr = 1.0_${k1}$ / r
+        ${t1}$ ::  x, y, re
+        integer :: hz, iz, i
+
+        if(scale == 0._${k1}$) call error_stop("Error(rvs_norm_array): Normal" &
+            //"distribution scale parameter must be non-zero")
+        if(.not. zig_norm_initialized) call zigset
+        do i = 1, array_size
+            iz = 0
+            hz = dist_rand(1_int32)
+            iz = iand( hz, 127 )
+            if( abs( hz ) < kn(iz) ) then
+                re = hz * wn(iz)
+            else
+                L1: do
+                    L2: if( iz == 0 ) then
+                        do
+                            x = -log( uni(1.0_${k1}$) ) * rr
+                            y = -log( uni(1.0_${k1}$) )
+                            if( y + y >= x * x ) exit
+                        end do
+                        re = r + x
+                        if( hz <= 0 ) re = -re
+                        exit L1
+                    end if L2
+                    x = hz * wn(iz)
+                    if( fn(iz) + uni(1.0_${k1}$) * (fn(iz-1) - fn(iz)) <        &
+                        exp(-HALF * x * x) ) then
+                        re = x
+                        exit L1
+                    end if
+
+                    hz = dist_rand(1_int32)
+                    iz = iand( hz, 127 )
+                    if( abs( hz ) < kn(iz) ) then
+                        re = hz * wn(iz)
+                        exit L1
+                    end if
+                end do L1
+            end if
+            res(i) = re * scale + loc
+        end do
+    end function rvs_norm_array_${t1[0]}$${k1}$
+
+    #:endfor
+
+
+
+    #:for k1, t1 in CMPLX_KINDS_TYPES
+    function rvs_norm_array_${t1[0]}$${k1}$(loc, scale, array_size) result(res)
+        ${t1}$, intent(in) :: loc, scale
+        integer, intent(in) :: array_size
+        integer :: i
+        ${t1}$ :: res(array_size)
+        real(${k1}$) :: tr, ti
+
+        do i = 1, array_size
+            tr = rvs_norm_r${k1}$(loc % re, scale % re)
+            ti = rvs_norm_r${k1}$(loc % im, scale % im)
+            res(i) = cmplx(tr, ti, kind=${k1}$)
+        end do
+    end function rvs_norm_array_${t1[0]}$${k1}$
+
+    #:endfor
+
+
+
+    #:for k1, t1 in REAL_KINDS_TYPES
+    impure elemental function pdf_norm_${t1[0]}$${k1}$(x, loc, scale) result(res)
+    !
+    ! Normal distribution probability density function
+    !
+        ${t1}$, intent(in) :: x, loc, scale
+        real :: res
+        ${t1}$, parameter :: sqrt_2_pi = sqrt(2.0_${k1}$ * acos(-1.0_${k1}$))
+
+        if(scale == 0._${k1}$) call error_stop("Error(pdf_norm): Normal"       &
+            //"distribution scale parameter must be non-zero")
+        res = exp(- 0.5_${k1}$ * (x - loc) * (x - loc) / (scale * scale)) /    &
```

Comment:
Here, to reduce the risk of numerical problems when `x, loc, scale` are all very large magnitude, perhaps it would be better to write this as `exp(- 0.5_${k1}$ * ((x - loc)/scale) * ((x - loc) /scale))) `

Reply:
@Jim-215-Fisher what do you think about this change?

Reply:
Sorry, did not see it.